### PR TITLE
Feature/optimize sentinel elect

### PR DESCRIPTION
**Pull request by Funkydream**
1 chang elect logic, use leader_epoch instead of current_peoch
2 flush config will not load from disk which will use avg 8ms and cause tilt
3 optimize parameters

### CODE REVIEW

File: src/sentinel.c [hunk 1]
```diff
@@ -5095,9 +5114,18 @@ void sentinelCheckTiltCondition(void) {
     sentinel.previous_time = mstime();
 }
 
+void sentinelCheckFlush(void) {
```
Comment (patpatbear) [inline src/sentinel.c:5117]: sentinelFlushConfigIfNeeded

File: src/config.c [hunk 2]
```diff
@@ -1141,80 +1141,83 @@ void rewriteConfigMarkAsProcessed(struct rewriteConfigState *state, const char *
  * If it is impossible to read the old file, NULL is returned.
  * If the old file does not exist at all, an empty state is returned. */
 struct rewriteConfigState *rewriteConfigReadOldFile(char *path) {
```
[the remark this answers is not in the document]
Comment (Funkydream) [reply src/config.c:1143]: 已经修改，仍然读磁盘，只是Sentinel在voteLeader时不FlushConfig

File: src/sentinel.c [hunk 3]
```diff
@@ -4439,19 +4433,27 @@ char *sentinelGetLeader(sentinelRedisInstance *master, uint64_t epoch) {
...
     uint64_t leader_epoch;
+    uint64_t counter_epoch;
     uint64_t max_votes = 0;
 
     serverAssert(master->flags & (SRI_O_DOWN|SRI_FAILOVER_IN_PROGRESS));
     counters = dictCreate(&leaderVotesDictType,NULL);
 
     voters = dictSize(master->sentinels)+1; /* All the other sentinels and me.*/
 
+    serverLog(LL_DEBUG,"helper: start vote master | master :%s, master epoch %llu, master->leader_epoch %llu",
```
[the remark this answers is not in the document]
Comment (Funkydream) [reply src/sentinel.c]: 类似情况已经修改

File: src/sentinel.c (hunk 1 above)
Comment (Funkydream) [reply src/sentinel.c]: 已经修改

**Pull request merged**